pkg/querier/unified_querier.go

```diff
@@ -26,7 +26,16 @@ func newUnifiedChunkQueryable(ds, cs ChunkStore, distributor Distributor, chunkI
 				mint:              mint,
 				maxt:              maxt,
 			},
-		}, nil
+		}
+
+		// Include ingester only if maxt is within 2 times ingester chunk age w.r.t. current time.
+		if maxt >= time.Now().Add(-2*ingesterMaxChunkAge).UnixNano()/1e6 {
+			ucq.stores = []ChunkStore{ds, cs}
+		} else {
+			ucq.stores = []ChunkStore{cs}
```

Comment:
Ditto, skip the else and use append.  And initialise `ucq.stores` to `[]ChunkStore{cs}`.